release-calendar/src/client/models/release-channel.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Release} from '../../types';
+
+export function getChannelRTVs(releases: Release[]): string[] {
+  const channelRTVs: string[] = [];
+  channelRTVs.push(releases.find(release => release.channel == 'lts').name);
+  channelRTVs.push(releases.find(release => release.channel == 'stable').name);
+  channelRTVs.push(
+    releases.find(release => release.channel == 'perc-beta').name,
+  );
+  channelRTVs.push(
+    releases.find(release => release.channel == 'perc-experimental').name,
+  );
+  channelRTVs.push(
+    releases.find(release => release.channel == 'opt-in-beta').name,
+  );
+  channelRTVs.push(
+    releases.find(release => release.channel == 'opt-in-experimental').name,
+  );
+  channelRTVs.push(releases.find(release => release.channel == 'nightly').name);
```

Comment:
So this returns a sorted array by channel? Do you see any potential bugs this could cause?

Reply:
I was making the assumption that when I query the releases that I can order them my date. If youre thinking that assumption isn't something I can make then yeah that could be a big bug. Or do you mean that when I push onto the channelRTVs array. From what I can find online, `.push()` always adds to the last index. Is that not true?

Reply:
This comment has to do the the push. The logic that `array[x]` gives the `y` channel is not maintainable because `array` can be updated at any time, and the developer would have to know somehow that the `x` index means `y` channel.

There should be other data structures that work better than an array.

Reply:
Oh, understood! might do a map again then :)